src/main/decompInitMod.F90

```diff
@@ -306,6 +257,174 @@ subroutine decompInit_lnd(lni, lnj, amask)
+            return
+         end if
+
+      end subroutine decompInit_lnd_allocate
+
+      !------------------------------------------------------------------------------
+
+      subroutine decompInit_lnd_gindex_global_allocate( bounds, ier )
+         ! Allocate gindex_global which requires that bounds gridcell begg to endg be set first
+         integer, intent(out) :: ier ! error code
+         type(bounds_type), intent(in) :: bounds ! contains subgrid bounds data
+
+         ier = 0
+         if ( bounds%endg < 1 )then
+            ier = 1
+            call endrun(msg="endg is NOT set before allocation", file=sourcefile, line=__LINE__)
+            return
+         end if
+         allocate(gindex_global(1:bounds%endg), stat=ier)
+         if (ier /= 0) then
+            call endrun(msg="allocation error for gindex_global", file=sourcefile, line=__LINE__)
+            return
+         end if
+      end subroutine decompInit_lnd_gindex_global_allocate
+
+      !------------------------------------------------------------------------------
+
+      subroutine decompInit_lnd_clean()
+         ! Deallocate the temporary variables used in decompInit_lnd
+         deallocate(clumpcnt)
+         deallocate(gdc2glo)
+         !deallocate(lcid)
+      end subroutine decompInit_lnd_clean
+
+      !------------------------------------------------------------------------------
+
+      subroutine decompInit_lnd_set_nclumps_numg_lns( )
+         ! Set nclumps, numg, and lns
+         ! Because of this -- it HAS to be called before decompInit_lnd_allocate
+
+         ! Set total clumps and total cells over grid
+         nclumps = clump_pproc * npes
+
+         lns = lni * lnj
+
+         ! count total land gridcells
+         numg = 0
+         do ln = 1,lns
+            if (amask(ln) == 1) then
+               numg = numg + 1
+            endif
+         enddo
+
+      end subroutine decompInit_lnd_set_nclumps_numg_lns
+
+      !------------------------------------------------------------------------------
+
+      subroutine decompInit_lnd_check_errors( ier )
+         ! Do some general error checking on input options
+         integer, intent(out) :: ier ! error code
+
+         ier = 0
+         if (nsegspc < 1) then
+            ier = 1
+            write(iulog,*) 'nsegspc bad = ',  nsegspc
+            call endrun(msg="Number of segments per clump (nsegspc) is less than 1 and can NOT be", &
+                        file=sourcefile, line=__LINE__)
+            return
+         end if
+
+         !--- set and verify nclumps ---
+         if (clump_pproc > 0) then
+            if (nclumps < npes) then
+               ier = 1
+               write(iulog,*) 'Number of gridcell clumps= ',nclumps, &
+                     ' is less than the number of processes = ', npes
+               call endrun(msg="Number of clumps exceeds number of processes", &
+                           file=sourcefile, line=__LINE__)
+               return
```

Comment:
@ekluzek I am confused about these two messages. The latter one is says nclumps exceeds npes, but the condition says otherwise. Also, can it be PEs instead of processes? Does the rest of CESM use processes to describe threads*tasks?

Reply:
To prevent having to repeat testing, @ekluzek will address this concern in #3412

Reply:
@mvdebolskiy thanks so much for pointing this out! Yes, the description of the error reported is wrong (and has been wrong for years before this). I'll point you to the other PR where I'll fix this.